software/CONFIGURATION.md

```diff
@@ -53,3 +53,84 @@ shows the default configuration:
 ```
 
 - `volts_per_octave` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+
+
+# Accessing config members in Python code
+
+The firmware converts the JSON file into a `ConfigSettings` object, where the JSON keys are converted
+to Python attributes.  The JSON object's keys must follow these rules, otherwise a `ValueError` will be raised:
+
+1. The string cannot be empty
+1. The string may only contain letters, numbers, and the underscore (`_`) character
+1. The string may not begin with a number
+
+The JSON key is converted to upper-case and turned into a Python attribute of the configuration object. For example,
+this JSON file
+```json
+{
+  "clock_multiplier": 4,
+  "hard_sync": true,
+  "wave_shape: "sine"
```

Comment:
I'm not sure what's going on with the formatting here, but at least on my computer the `sine"` part of line 73 is highlighted in red in both the raw and rich diff

![image](https://github.com/Allen-Synthesis/EuroPi/assets/79809962/54527063-4445-4e49-92ac-52b54238ac4e)

Reply:
I've just noticed that this is due to the quote marks being missing at the end of `wave_shape`

Reply:
Fixed